kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/marshalling/xstream/XStreamMarshaller.java

```diff
@@ -143,86 +162,107 @@ private MapperWrapper chainMapperWrappers(List<XStreamMarshallerExtension> exten
+        xstream.processAnnotations(GetReleaseIdCommand.class);
+        xstream.processAnnotations(UpdateReleaseIdCommand.class);
+        xstream.processAnnotations(GetServerInfoCommand.class);
+        xstream.processAnnotations(ListContainersCommand.class);
+        xstream.processAnnotations(ServiceResponsesList.class);
+        xstream.processAnnotations(ServiceResponse.class);
+        xstream.processAnnotations(KieContainerResourceList.class);
+        xstream.processAnnotations(KieContainerResource.class);
+        xstream.processAnnotations(ReleaseId.class);
+        xstream.processAnnotations(KieContainerStatus.class);
+        xstream.processAnnotations(KieScannerResource.class);
+        xstream.processAnnotations(KieServerInfo.class);
+
+        xstream.processAnnotations(ReleaseIdFilter.class);
+        xstream.processAnnotations(KieContainerStatusFilter.class);
+        xstream.processAnnotations(KieContainerResourceFilter.class);
+
+        xstream.processAnnotations(SolverInstance.class);
+        xstream.processAnnotations(CreateSolverCommand.class);
+        xstream.processAnnotations(DisposeSolverCommand.class);
+        xstream.processAnnotations(GetSolverWithBestSolutionCommand.class);
+        xstream.processAnnotations(GetSolversCommand.class);
+        xstream.processAnnotations(GetSolverCommand.class);
+        xstream.processAnnotations(SolvePlanningProblemCommand.class);
+        xstream.processAnnotations(TerminateSolverEarlyCommand.class);
+        xstream.processAnnotations(AddProblemFactChangeCommand.class);
+        xstream.processAnnotations(AddProblemFactChangesCommand.class);
+        xstream.processAnnotations(IsEveryProblemFactChangeProcessedCommand.class);
+
+        xstream.processAnnotations(DMNContextKS.class);
+        xstream.processAnnotations(DMNResultKS.class);
+        xstream.processAnnotations(DMNNodeStub.class);
+        xstream.processAnnotations(DMNMessageKS.class);
+        xstream.processAnnotations(DMNDecisionResultKS.class);
+        xstream.processAnnotations(DMNModelInfoList.class);
+        xstream.processAnnotations(DMNModelInfo.class);
+        xstream.processAnnotations(DMNDecisionInfo.class);
+        xstream.processAnnotations(DMNInputDataInfo.class);
+        xstream.processAnnotations(DMNItemDefinitionInfo.class);
 
-        if (classes != null) {
-            for (Class<?> clazz : classes) {
-                this.xstream.processAnnotations(clazz);
-                this.classNames.put(clazz.getName(),
-                                    clazz);
-            }
-        }
     }
 
     @Override
     public String marshall(Object objectInput) {
         return xstream.toXML(objectInput);
     }
 
+    private XStream getProperXstream(Map<String, Object> parameters) {
+        Object ignoreString = parameters.get(XSTREAM_IGNORE_UNKNOWN);
+        if (ignoreString != null && Boolean.parseBoolean(ignoreString.toString())) {
+            ignoreLock.lock();
```

Comment:
Not sure if lazy initialization is the best aprroach here. 
We are certainly saving memory if nobody (as we expect) is using the parameter in a sunny day scenario. 
On the other hand, when the parameter is enable, threads execution will be serialized because of the lock. 
I think this is better because it turns out that xstream object is not really inmutable

Reply:
locks are not needed.